src/ripple/rpc/handlers/AccountLines.cpp

```diff
@@ -96,31 +96,29 @@ doAccountLines(RPC::JsonContext& context)
     if (!ledger)
         return result;
 
-    std::string strIdent(params[jss::account].asString());
-    AccountID accountID;
-
-    if (auto jv = RPC::accountFromString(accountID, strIdent))
+    std::string const strIdent(params[jss::account].asString());
+    auto const accountID = parseBase58<AccountID>(strIdent);
+    if (!accountID)
     {
-        for (auto it = jv.begin(); it != jv.end(); ++it)
```

Comment:
What did this old code do and why is it removed?

Reply:
The function [RPC::accountFromString()](https://github.com/XRPLF/rippled/blob/develop/src/ripple/rpc/impl/RPCHelpers.cpp#L85) attempts different ways to parse the input string (it checks whether the input is a public key, an account ID or a seed). If  all attempts fail, the function returns a Json object that contains an error code and an error message. The `for` loop traverses the return Json object and add all its fields (error code and error message) to the `Json` result object.

The new code calls the function [inject_error](https://github.com/XRPLF/rippled/blob/develop/src/ripple/protocol/ErrorCodes.h#L212) and achieves the same effect.